### ISSUE TITLE
Prefer find over detect within a Rails project

### ISSUE THREAD

**Issue by zedtux** [rubocop/rubocop#1334]
Given an ActiveRecord User model as the following:

``` ruby
class User < ActiveRecord::Base
  has_many :beer_packs

  def beer_pack_with_remaining_beer
    beer_packs.detect do |beer_pack|
      beer_pack.with_remaining_bottle?
    end
  end
end
```

Rubocop complain as the following:

```
app/models/user.rb:5:15: C: Prefer find over detect.
    beer_packs.detect do |beer_pack|
               ^^^^^^
```

In the case I would apply to use `find` instead of `detect` then my code wouldn't work.
Expected: When using the `-R` or `--rails`, when `detect` is used on an association it shouldn't raise an offence.
